tests/test_core.py

```diff
@@ -143,72 +144,70 @@ def test_find_dotenv():
     assert find_dotenv(usecwd=True) == ''
 
     # now place a .env file a few levels up and make sure it's found
-    filename = os.path.join(child1, '.env')
-    with open(filename, 'w') as f:
-        f.write("TEST=test\n")
-    assert find_dotenv(usecwd=True) == filename
+    dotenv_file = child1 / '.env'
+    dotenv_file.write_bytes(b"TEST=test\n")
+    assert find_dotenv(usecwd=True) == str(dotenv_file)
 
 
-def test_load_dotenv(cli):
+def test_load_dotenv(tmp_path):
+    os.chdir(str(tmp_path))
     dotenv_path = '.test_load_dotenv'
-    with cli.isolated_filesystem():
-        sh.touch(dotenv_path)
-        set_key(dotenv_path, 'DOTENV', 'WORKS')
-        assert 'DOTENV' not in os.environ
-        success = load_dotenv(dotenv_path)
-        assert success
-        assert 'DOTENV' in os.environ
-        assert os.environ['DOTENV'] == 'WORKS'
-        sh.rm(dotenv_path)
-
-
-def test_load_dotenv_override(cli):
+    sh.touch(dotenv_path)
+    set_key(dotenv_path, 'DOTENV', 'WORKS')
+    assert 'DOTENV' not in os.environ
+    success = load_dotenv(dotenv_path)
+    assert success
+    assert 'DOTENV' in os.environ
+    assert os.environ['DOTENV'] == 'WORKS'
+
+
+def test_load_dotenv_override(tmp_path):
+    os.chdir(str(tmp_path))
     dotenv_path = '.test_load_dotenv_override'
     key_name = "DOTENV_OVER"
-
-    with cli.isolated_filesystem():
-        sh.touch(dotenv_path)
-        os.environ[key_name] = "OVERRIDE"
-        set_key(dotenv_path, key_name, 'WORKS')
-        success = load_dotenv(dotenv_path, override=True)
-        assert success
-        assert key_name in os.environ
-        assert os.environ[key_name] == 'WORKS'
-        sh.rm(dotenv_path)
-
-
-def test_load_dotenv_in_current_dir():
-    # make sure were are here!
-    os.chdir(os.path.dirname(os.path.realpath(__file__)))
-    dotenv_path = '.env'
-    with open(dotenv_path, 'w') as f:
-        f.write("TOTO=bla\n")
-    assert 'TOTO' not in os.environ
-    success = load_dotenv(verbose=True)
+    sh.touch(dotenv_path)
+    os.environ[key_name] = "OVERRIDE"
+    set_key(dotenv_path, key_name, 'WORKS')
+    success = load_dotenv(dotenv_path, override=True)
     assert success
-    assert os.environ['TOTO'] == 'bla'
-    sh.rm(dotenv_path)
+    assert key_name in os.environ
+    assert os.environ[key_name] == 'WORKS'
+
+
+@pytest.mark.xfail(sys.version_info < (3, 0), reason="test was incomplete")
```

Comment:
@smsearcy This updated test shows the 2.7 dotenv detection bug on Travis (couldn't reproduce it on my machine though).  Your fix in #152 makes it pass, which is a good sign.

Reply:
What's odd is that reviewing the coverage changes for #152 after merging #154 isn't showing it covering that code, and yet now the test passes.  I'll dig into this a little more.

Reply:
The new test unfortunately doesn't use the same Python VM, and so doesn't contribute to coverage.  But this is also what makes this test isolated and realistic.